### ISSUE TITLE
Resource instance usage summary report endpoint is currently unusable without the SAMPLING property configured

### ISSUE THREAD

**Issue by astefanova** [cloudfoundry-incubator/cf-abacus#559]
The following [endpoint](https://github.com/cloudfoundry-incubator/cf-abacus/blob/f30e11a640614c8409087af805ccb741ab871b63/doc/api.md#resource-instance-usage-summary-report):
```
GET /v1/metering/organizations/:organization_id/resource_instances/:resource_instance_id/consumers/:consumer_id/plans/:plan_id/metering_plans/:metering_plan_id/rating_plans/:rating_plan_id/pricing_plans/:pricing_plan_id/t/:t/aggregated/usage/:time
```
 is currently unusable, because the user cannot know the `:t` parameter (without checking it in the DB). 

The same issue applies for the graphql query in the last example in the [doc](https://github.com/cloudfoundry-incubator/cf-abacus/blob/f30e11a640614c8409087af805ccb741ab871b63/doc/api.md#example-graphql-queries).

With the SAMPLING property configured, we can guess more accurately, but we should probably find a way to make it usable even without that property.

**Comment by rajkiranrbala** [comment]
@astefanova

The order to fetch the reports is you get the org report, which included the consumer reports. The consumer reports contain the resource instance id and time `:t` parameter. In this way, we do not have to guess the resource instance documents.

**Comment by hsiliev** [comment]
I can see the following on consumer level when getting org usage report:
```
consumers: 
        [ { consumer_id: 'app:bbeae239-f3f8-483c-9dd0-de6781c38bab',
            resources: 
             [ { resource_id: 'object-storage',
                 plans: 
                  [ { plan_id: 'basic/basic-object-storage/object-rating-plan/object-pricing-basic',
                      aggregated_usage: 
                       [ { metric: 'storage',
                           windows: { charge: 2, summary: 2, cost: 2, quantity: 2 } },
                         { metric: 'thousand_light_api_calls',
                           windows: { charge: 0.27, summary: 9, cost: 0.27, quantity: 9 } },
                         { metric: 'heavy_api_calls',
                           windows: { charge: 135, summary: 900, cost: 135, quantity: 900 } } ],
                      resource_instances: 
                       [ { id: '0b39fa70-a65f-4183-bae8-385633ca5c87',
                           t: '0001488491362788-0-0-1-0',
                           p: 1488491362788 } ],
                      metering_plan_id: 'basic-object-storage',
                      rating_plan_id: 'object-rating-plan',
                      pricing_plan_id: 'object-pricing-basic',
                      windows: { charge: 137.27 } } ],
                 aggregated_usage: 
                  [ { metric: 'storage', windows: { charge: 2 } },
                    { metric: 'thousand_light_api_calls',
                      windows: { charge: 0.27 } },
                    { metric: 'heavy_api_calls', windows: { charge: 135 } } ],
                 windows: { charge: 137.27 } } ],
            windows: { charge: 137.27 } } ],
       windows: { charge: 137.27 } } ],

```
The needed info is located here:
```
                   resource_instances: 
                       [ { id: '0b39fa70-a65f-4183-bae8-385633ca5c87',
                           t: '0001488491362788-0-0-1-0',
                           p: 1488491362788 } ],
```

Closing the issue

**Comment by KRuelY** [comment]
To elaborate the reasoning behind this design:

Fetching detailed org report would be inefficient if the doc is huge. The resource instance and consumer report is the one that contributed to huge org report size, so we decided to have a small section containing list of consumers and resource instances.

The information needed to get consumer report would be in org report, and the information needed for instance report would be in consumer report. This was implemented this way, so that the UI on top of the org report can be more efficient. It would only fetch consumer report / instance report when the user want to see it rather than display everything at once.
